src/test/java/org/javarosa/core/model/FormDefSerializationTest.java

```diff
@@ -77,6 +80,32 @@ public class FormDefSerializationTest {
         assertThat(deserialized.getFormDef().getMainInstance().getBase().getInstanceName(), is(nullValue()));
     }
 
+    @Test public void serializeAndDeserializeFormWithDefaultAnswerInSelectQuestion_worksCorrectly() throws IOException, DeserializationException, XFormParser.ParseException {
+        Scenario scenario = Scenario.init("Internal choices", html(
+            head(
+                title("Internal choices"),
+                model(
+                    mainInstance(
+                        t("data id='main-instance'",
+                            t("select-from-secondary-instance", "a")
+                        )
+                    ),
+                    instance("secondary-instance",
+                        t("item",
+                            t("label", "A"),
+                            t("value", "a")
+                        )
+                    )
+                )
+            ),
+            body(
+                select1Dynamic("/data/select-from-secondary-instance", "instance('secondary-instance')/root/item")
+            ))
+        );
+
+        scenario.serializeAndDeserializeForm();
```

Comment:
It would be ideal to have a test that fails when checking the instance name or reference of the `select_one` question. However, the issue occurs only after deserialization, and before any validation can be performed, the deserialization process throws an exception.

Reply:
This is an integration test and that feels ok. I wish the test didn't end with `_worksCorrectly` but I also don't have a great alternate suggestion so I'm inclined to go with it!